source/Program.cs

```diff
@@ -14,45 +19,95 @@ class Program
     {
         public static async Task Main(string[] args)
         {
-            var host = Host.CreateDefaultBuilder(args)
-                .ConfigureAppConfiguration((_, config) =>
-                {
-                    config.AddEnvironmentVariables();
-                })
-                .ConfigureServices((_, services) =>
-                {
-                    services.AddMetricFactory();
-                    services.AddSingleton<IMetricServer>(sp => new MetricServer(
-                        new MetricServerOptions
-                        {
-                            CollectorRegistryInstance = sp.GetRequiredService<ICollectorRegistry>(),
-                            Host = "0.0.0.0",
-                            Port = 9090,
-                            UseDefaultCollectors = false
-                        }));
-                    services.AddHostedService<TeamCityQueueLengthScraper>();
-                    services.AddHostedService<TeamCityBuildScraper>();
-                    services.AddHostedService<TeamCityBuildArtifactScraper>();
-                    services.AddHostedService<TeamCityQueueWaitScraper>();
-                })
-                .UseConsoleLifetime()
-                .Build();
+            LogUnhandledExceptions();
+            LogUnobservedTaskExceptions();
+            
+            var configBuilder = new ConfigurationBuilder()
+                .SetBasePath(Directory.GetCurrentDirectory())
+                .AddEnvironmentVariables();
 
-            var metricServer = host.Services.GetRequiredService<IMetricServer>();
+            var config = configBuilder.Build();
+            var levelSwitch = new LoggingLevelSwitch(LogEventLevel.Debug); //allows us to switch log levels by modifying the seq
 
+            var appVersion = typeof(Program).Assembly?.GetName()?.Version?.ToString() ?? "<unknown>";
+            var loggerConfiguration = new LoggerConfiguration()
+                .MinimumLevel.Debug()
+                .MinimumLevel.Override("Microsoft", LogEventLevel.Information)
+                .MinimumLevel.Override("Microsoft.AspNetCore", LogEventLevel.Warning)
+                .Enrich.FromLogContext()
+                .Enrich.WithProperty("Environment", Environment.GetEnvironmentVariable("ASPNET_ENVIRONMENT") ?? "Production")
```

Comment:
This is unlikely to be present in the test nautilus cluster (if we even run it there???)

Reply:
I've modified - we'll add an env var `DOTNET_ENVIRONMENT` to the deployment to specify it